### PR TITLE
Scoreboard improvements

### PR DESCRIPTION
**Pull request by Tim203**
Fixed some Scoreboard bugs and ScoreboardUpdater is now global

### CODE REVIEW

File: connector/src/main/java/org/geysermc/connector/scoreboard/ScoreboardUpdater.java [hunk 1]
```diff
@@ -133,11 +142,11 @@ private long getTimeTillNextAction() {
 
     @RequiredArgsConstructor
     @Getter
-    public static class ScoreboardSession {
+    public static final class ScoreboardSession {
         private final GeyserSession session;
         private final AtomicInteger pendingPacketsPerSecond = new AtomicInteger(0);
         private int packetsPerSecond;
-        private int millisBetweenUpdates = FIRST_MILLIS_BETWEEN_UPDATES;
+        private boolean justSwitchedToSelf;
```
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/scoreboard/ScoreboardUpdater.java:149]: Can you add a Javadoc comment here explaining what this is for?

Comment (Tim203) [reply connector/src/main/java/org/geysermc/connector/scoreboard/ScoreboardUpdater.java:149]: I tried to explain the purpose of it on line 87

File: connector/src/main/java/org/geysermc/connector/network/translators/java/scoreboard/JavaScoreboardObjectiveTranslator.java [hunk 2]
```diff
@@ -25,30 +25,38 @@
...
         WorldCache worldCache = session.getWorldCache();
         Scoreboard scoreboard = worldCache.getScoreboard();
-        Objective objective = scoreboard.getObjective(packet.getName());
         int pps = worldCache.increaseAndGetScoreboardPacketsPerSecond();
 
-        if (objective == null && packet.getAction() != ObjectiveAction.REMOVE) {
-            objective = scoreboard.registerNewObjective(packet.getName(), false);
+        Objective objective = scoreboard.getObjective(packet.getName());
+        if (objective != null && objective.getUpdateType() != UpdateType.REMOVE && packet.getAction() == ObjectiveAction.ADD) {
+            logger.warning("An objective with the same name '" + packet.getName() + "' already exists! Ignoring packet");
```
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/network/translators/java/scoreboard/JavaScoreboardObjectiveTranslator.java:54]: If this matches Java Edition logging, please add a comment saying so.

File: connector/src/main/java/org/geysermc/connector/scoreboard/Scoreboard.java [hunk 3]
```diff
@@ -42,14 +42,15 @@
...
-@Getter
 public final class Scoreboard {
     private final GeyserSession session;
     private final GeyserLogger logger;
+    @Getter
     private final AtomicLong nextId = new AtomicLong(0);
 
-    private final Map<String, Objective> objectives = new ConcurrentHashMap<>();
-    private final Map<String, Team> teams = new HashMap<>();
+    private final Map<String, Objective> objectives = new HashMap<>();
```
Comment (Camotoy) [inline connector/src/main/java/org/geysermc/connector/scoreboard/Scoreboard.java:51]: This appears to be accessed on multiple threads (both the translators and the onUpdate).

**Review by Camotoy — APPROVED** [review]

**Pull request merged**